src/shared_modules/utils/windowsHelper.h

```diff
@@ -136,6 +169,54 @@ namespace Utils
         return ret;
     }
 
+
+    /* Reference: https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_2.6.0.pdf */
+    static std::string parseRawSmbios(const BYTE* rawData, const DWORD rawDataSize)
+    {
+        std::string serialNumber;
+        DWORD offset{0};
+
+        while (offset < rawDataSize && serialNumber.empty())
+        {
+            SMBIOSStructureHeader header{};
+            memcpy(&header, rawData + offset, sizeof(SMBIOSStructureHeader));
```

Comment:
This instruction is assuming that `rawData + offset` always has  `sizeof(SMBIOSStructureHeader)` remaining bytes to copy into `header` and that might not be the case.

Consider the same for `memcpy(&info, rawData + offset, sizeof(SMBIOSBasboardInfoStructure));`